nixos/modules/tasks/auto-upgrade.nix

```diff
@@ -63,13 +63,16 @@ in {
       };
 
       dates = mkOption {
-        default = "04:40";
         type = types.str;
+        default = "daily";
```

Comment:
Why change the old default?

But if we do need to change it anyway, I think this may needs some documentation in the changelog for the next release :thinking: ?

Reply:
Because of security. Currently if you just enable auto-upgrade it will never run on laptop (unless you keep it running over night, but who does it with laptops?). So user might expect to get security fixes, but he doesn't.

Reply:
I thought just setting `persistent = true` fixed this issue. Does it not?

Reply:
Well, I read systemd manual, but I am not sure if persistent applies to a specific date/time. It looked it applies only to intervals, but I might be wrong. Do you know it better?

Reply:
Can confirm it does apply to specific date/times based on a month of using it in my config.

Reply:
Ok, I'll revert to original specific date/times. One last question, if set to 4:40, that is very early morning, does it mean that for laptop it always tries to run it as soon as it finishes booting? Would like to know how it behaves with daily. I'd like to avoid every such timer expiring right after boots finishes... Would be better to spread it out more... But eh, probably just trying to find good spot where there are none, right?

Reply:
I think the manpage give us the answer:

```
Persistent=

    Takes a boolean argument. If true, the time when the service unit was last triggered is stored on disk. When the timer is activated, the service unit is triggered immediately if it would have been triggered at least once during the time when the timer was inactive. Such triggering is nonetheless subject to the delay imposed by RandomizedDelaySec=. This is useful to catch up on missed runs of the service when the system was powered down. Note that this setting only has an effect on timers configured with OnCalendar=. Defaults to false.

    Use systemctl clean --what=state … on the timer unit to remove the timestamp file maintained by this option from disk. In particular, use this command before uninstalling a timer unit. See systemctl(1) for details.
```

Reply:
I'll return defaults back to original value. Do you agree?